bin/kafka-run-class.sh

```diff
@@ -22,6 +22,12 @@ fi
 
 base_dir=$(dirname $0)/..
 
+# run kafka-env.sh
+KAFKA_ENV=$base_dir/config/kafka-env.sh
```

Comment:
Good points, Ewen. As always :)

I think sharing one file is fine. Since the file can be left empty and whatever you currently use to set environment variables will still work.
I believe that in a production environment all these processes (kafka, zookeeper, clients) will run on separate machines and therefore won't use the same environment file.

I am more concerned about the fact that kafka-server-state.sh (and possibly others) now check for environment variables and possibly set defaults that later get overwritten. 

I'd expect that:
1. Everything I set with export will take priority.
2. All environment defaults will be set in the file. The way some defaults are defined (and then maybe overwritten) in other scripts is kinda inconsistent and confusing.

Does that make sense? It is possible that my expectations are all wrong :)